docs/troubleshooting-guide/sql-error-message.md

```diff
@@ -0,0 +1,199 @@
+      - ` SELECT {% raw %}``column_name``{% endraw %} `
+
+    - Are you using the correct path to columns and tables? For example:
+
+      - `FROM table_name`
+      - `FROM schema_name.table_name`
+      - `FROM database_name.schema_name.table_name`
+
+    - Is your column name a reserved word? For example:
+
+      [In PostgresSQL, 'users' is a reserved key word](https://www.postgresql.org/docs/current/sql-keywords-appendix.html).
+
+      - `SELECT users` will throw an error.
+      - `SELECT "users"` will run correctly.
+
+2.  Review the [data reference][data-reference] for the tables and columns in your query.
+
+    - Does the column name exist in the table?
+
+    - If the column name exists, but you can’t query the column from the SQL editor:
+      - Ask your Metabase admin if the column was re-named or removed on the database side.
+      - If you’re a Metabase admin, you may need to [run a sync][database-syncing] to refresh your data.
+
+3.  If you no longer have an error message, but your query results are incorrect, go to [Troubleshooting SQL logic][troubleshoot-sql-logic].
+4.  If you're still stuck, search or ask the community on [Discourse][discourse].
+
+**Causes:**
+
+- You need to use the correct SQL syntax for the database that stores the tables you want to query.
+- If a column or table name was updated in your database, but your Metabase hasn’t run a sync with your database yet, your database won’t recognize the names from your query.
+
+**Further reading:**
+
+- [How Metabase executes SQL queries][how-metabase-executes-queries].
+- [How Metabase syncs with your database][database-syncing].
+
+#### My SQL "function does not exist”.
+
+If your SQL query contains [SQL variables][sql-variable-def] that look like `{% raw %}{{ variable }}{% endraw %}`, go to [Troubleshooting SQL variables][troubleshoot-sql-variables] first.
+
+**Steps:**
+
+1. Review the data type of the column that you want your function to apply to.
+
+   - You can use the Metabase [data reference][data-reference] to review the column's [field type][field-type-def] (as a proxy for [data type][data-type-def]).
+   - You can also directly query the information schema in your database if you have permission to access it.
+
+2. Review the **function** section of the [reference guide][sql-reference-guide] for your SQL dialect.
+
+   - Confirm that the function exists for your SQL dialect.
+   - Review the data type(s) that are accepted by your function.
+
+3. If the field type of your column does not match the expected data type of your function:
+
+   - Cast your column to the correct data type in your SQL query.
+   - If you’re a Metabase admin, you can also [cast data types from the Data model page][how-to-cast-data-type].
+
+4. If you no longer have an error message, but your query results are incorrect, go to [Troubleshooting SQL logic][troubleshoot-sql-logic].
+5. If you're still stuck, search or ask the community on [Discourse][discourse].
+
+**Causes:**
```

Comment:
Again, is the list the best format for this? Right now the list items are of different types: 1) explanation, 2) example 2) note. Maybe use a paragraph(s)?